tests/backends/pyq/test_quantum_pyq.py

```diff
@@ -844,3 +846,12 @@ def test_move_to_dtype(
     assert wf.dtype == dtype
     expval = qm.expectation(inputs, state=state)
     assert expval.dtype == torch.float64 if dtype == torch.cdouble else torch.float32
+
+
+
+@pytest.mark.parametrize("gen", [Z(0) + Z(1), Z(0) + Z(1) + X(0) - X(1), Z(0) + Z(1) + X(0) - X(0)])
+def test_hamiltonian_evolution(gen: AbstractBlock) -> None:
+    state = run(HamEvo(gen, torch.tensor([1.0])))
+    assert type(state) == torch.Tensor
+    assert state.dtype == torch.complex128
+    assert not torch.any(torch.isnan(state))
```

Comment:
No need to include this as a test, there are already many tests that run `HamEvo`.

Also it seems this is conflicting with the main branch. Probably it will be solved once you remove the code here, but try to rebase anyway.

Reply:
Ok, it will be also removed